src/plugins/Network.js

```diff
@@ -470,6 +486,70 @@ module.exports = class Network extends OverlayPlugin {
                 option.selected = 'selected';
             }
         }
-        this.changeProfile();
+    }
+
+    /**
+     * Update mobile profile list UI according to the current active profile.
+     *
+     * @param {string} profile Profile name.
+     */
+    setActiveMobileProfile(profile) {
+        if (!profile) {
+            log.error('setActiveMobileProfile: Error : provided profile is empty');
+            return;
+        }
+        const mobileProfile = MOBILE_PROFILES.find((elem) => elem.name === profile);
+        if (!mobileProfile) {
+            return;
+        }
+
+        const options = this.select.getElementsByTagName('option');
+        for (let i = 0; i < options.length; i++) {
+            const option = options[i];
+            if (option.value === mobileProfile.name) {
+                option.selected = 'selected';
+            }
+        }
+    }
+
+    /**
+     * Update mobile signal strength list UI according to the current active strength.
+     *
+     * @param {string} strength Signal strength name.
+     */
+    setActiveSignalStrength(strength) {
+        if (!strength) {
+            return;
+        }
+        const signalStrength = MOBILE_SIGNAL_STRENGTH.find((elem) => elem.name === strength);
+        if (!signalStrength) {
+            return;
+        }
+
+        const options = this.selectMobileSignalStrength.getElementsByTagName('option');
+        for (let i = 0; i < options.length; i++) {
+            const option = options[i];
+            if (option.value === signalStrength.name) {
+                option.selected = 'selected';
+            }
+        }
+    }
+
+    /**
+     * Update mobile signal Detail information.
+     *
+     * @param {string} detail Signal detail to update.
+     * @param {string} value  New signal detail value.
+     * @param {string} reset  If true ignore value and set "".
+     */
+    updateDetail(detail, value, reset) {
```

Comment:
I think the `detail` argument will always be a defined string, no? Do we need this check `if(!detail)`?